src/Domain/BlockLogicalCoordinates.cpp

```diff
@@ -143,11 +141,23 @@ block_logical_coordinates_single_point(
 }
 
 template <size_t Dim, typename Frame>
-std::vector<block_logical_coord_holder<Dim>> block_logical_coordinates(
+std::vector<BlockLogicalCoords<Dim>> block_logical_coordinates(
     const Domain<Dim>& domain, const tnsr::I<DataVector, Dim, Frame>& x,
-    const double time, const functions_of_time_type& functions_of_time) {
+    const double time, const domain::FunctionsOfTimeMap& functions_of_time,
+    const std::optional<size_t> num_threads) {
+  // Resolve number of threads to use in OpenMP parallelization
+#ifdef _OPENMP
+  const size_t resolved_num_threads =
+      num_threads.value_or(omp_get_max_threads());
+#else
+  if (num_threads.has_value()) {
+    ERROR("OpenMP is not available, so num_threads cannot be specified.");
+  }
+#endif  // _OPENMP
+
   const size_t num_pts = get<0>(x).size();
-  std::vector<block_logical_coord_holder<Dim>> block_coord_holders(num_pts);
+  std::vector<BlockLogicalCoords<Dim>> block_coord_holders(num_pts);
+#pragma omp parallel for num_threads(resolved_num_threads)
   for (size_t s = 0; s < num_pts; ++s) {
```

Comment:
Did you ever try just doing:
```cpp
#include <thread>

const auto lambda = [&x, time, &functions_of_time](const size_t start, const size_t end) {
  for (size_t s = start; s < end; ++s) {
    tnsr::I<double, Dim, Frame> x_frame(0.0);
    for (size_t d = 0; d < Dim; ++d) {
      x_frame.get(d) = x.get(d)[s];
    }
    // Check which block this point is in. Each point will be in one
    // and only one block, unless it is on a shared boundary.  In that
    // case, choose the first matching block (and this block will have
    // the smallest block_id).
    for (const auto& block : domain.blocks()) {
      std::optional<tnsr::I<double, Dim, ::Frame::BlockLogical>> x_logical =
          block_logical_coordinates_single_point(x_frame, block, time,
                                                 functions_of_time);
      if (x_logical.has_value()) {
        // Point is in this block.  Don't bother checking subsequent
        // blocks.
        block_coord_holders[s] = make_id_pair(domain::BlockId(block.id()),
                                              std::move(x_logical.value()));
        break;
      }
    }
  }
};

if (num_threads.value_or(0) == 0) {
  lambda(0, num_pts);
} else {
  std::vector<std::thread> threads(num_threads.value());
  // Spawn threads to do the interpolation.
  for (auto& thread : threads) {
    // Note: you'll want to do the math so that the remainder grid points are approximately evenly distributed.
    thread = std::thread{lambda, this_thread_start, this_thread_end};
  }
  // Wait for threads to finish the interpolation.
  for (auto& thread : threads) {
    thread.join();
  }
}
```
I'm not sure if you'll get "perfect strong scaling", and it's more than "just a pragma", but it also doesn't require _any_ changes to other files, no env vars, etc. Plus, this doesn't even require a compiler that supports OpenMP (for a long time Clang didn't!)

Reply:
Thanks for offering the suggestion! Seems very reasonable. OpenMP is so easy though :D The parallelization of `block_logical_coordinates` is indeed just one line ("as advertised"). The parallelization of the exporter lib is a bit more complicated (still very easy though), but that's a separate lib that's only used for pybindings and exporter lib.

I limited linking OpenMP out of an abundance of caution before. I think that's still possible by compiling a parallelized version of `block_logical_coordinates` into the exporter lib, and leaving the domain lib untouched. I can try that.

Note that this parallelization speeds up the exporter by a factor 10 at least, and this is actually very noticeable when interpolating data for plots or analysis, and essential for importing data into another code. That's why I'd like to find a path for merging this.

I think independent of this we should disable OpenMP multithreading in our executables, so no dependencies accidentally try to do multithreading (like OpenBLAS, which we already disable, or libsharp). I think I'll just add a line `OMP_NUM_THREADS=1` to our submit script template, just to be safe.

Reply:
As noted above I think it's possible to remove the OpenMP linking from the domain lib (and thus also from the PCH and the rest of the code). I may not have written this clearly above. I can move the loop with its parallelization into the exporter lib and only call into the domain lib in the body of the loop. Then there's no parallelization in the domain lib at all.

Reply:
I think if you restrict the OMP pragmas to only the exporter lib it will be fine. Given the way Kokkos works, I will likely have Kokkos run in OpenMP mode while I debug our implementation before trying to run it on a GPU. That means if _any_ code that we use in our executables uses OpenMP, there's a problem. That means the domain files cannot have any OpenMP includes or directives, etc. in them.